includes/course-theme/class-sensei-course-theme-quiz.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * File containing Sensei_Course_Theme_Quiz class.
+ *
+ * @package sensei-lms
+ * @since 3.15.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Sensei_Course_Theme_Quiz class.
+ */
+class Sensei_Course_Theme_Quiz {
+
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_Course_Theme_Quiz constructor. Prevents other instances from being created outside of `self::instance()`.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		if ( 'quiz' !== get_post_type() ) {
+			return;
+		}
+
+		$this->maybe_add_quiz_results_notice();
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function maybe_add_quiz_results_notice() {
```

Comment:
This could be `private` too, right?

Reply:
Yes! Updated. cadcc75ce19908282a81694a786eb35dbcf5d853